odl/space/space_utils.py

```diff
@@ -22,14 +22,27 @@
 from future import standard_library
 standard_library.install_aliases()
 
-__all__ = ('vector',)
+__all__ = ('vector', 'ntuples', 'fn', 'cn', 'rn',
+           'NTUPLES_IMPLS', 'FN_IMPLS')
 
 import numpy as np
 
-from odl.space.ntuples import Rn, Cn, Fn, Ntuples
+from odl.set import RealNumbers, ComplexNumbers
+from odl.space.numpy_ntuples import NumpyNtuples, NumpyFn
 from odl.util.utility import (
     is_real_floating_dtype, is_complex_floating_dtype, is_scalar_dtype)
-from odl.space.cu_ntuples import CUDA_AVAILABLE, CudaRn, CudaFn, CudaNtuples
+from pkg_resources import iter_entry_points
+
+
+NTUPLES_IMPLS = {'numpy': NumpyNtuples}
+FN_IMPLS = {'numpy': NumpyFn}
+for entry_point in iter_entry_points(group='odl.space', name=None):
+    try:
+        module = entry_point.load()
+        NTUPLES_IMPLS.update(module.ntuples_impls())
+        FN_IMPLS.update(module.fn_impls())
+    except ImportError:
```

Comment:
> Not necessarily. In my case it made the tests print 500 MemoryError test failures, which hid the actual failing tests.

Well, `MemoryError` is also an error that needs to be fixed.

> That would be a viable solution.

I'll do that.